### ISSUE TITLE
Link citations?

### ISSUE THREAD

**Issue by vjeranc** [jaspervdj/hakyll#471]
Is it possible to get linked citations somehow with hakyll other than using unixFilter?

I can use the citations but I'd like for them to have hyperlinks (and possibly backlinks).

Currently I can get the wanted behavior using snippet below:

```
pandoc --bibliography bib --to=html5 --csl csl --filter pandoc-citeproc post.md
```

**Comment by jaspervdj** [comment]
You can use the [Hakyll.Web.Pandoc.Biblio](https://jaspervdj.be/hakyll/reference/Hakyll-Web-Pandoc-Biblio.html) module to do this.

**Comment by vjeranc** [comment]
Yes, I was not clear - forgot to add that the behavior I want works if `post.md` file contains the yaml header variable `link-citations: true` (when calling the snippet above).

Currently I'm using the Biblio constructs and do get working citations, but they don't have hyperlinks when cited in the text, they are just converted in the csl format I supply.

I can't see anywhere in the writer or reader options where could I add link-citations: true and get hyperlinks to references div.

**Comment by vjeranc** [comment]
I've checked the module you've linked and figured out a way to add the option.

It's a dirty fix but works for my needs. I'll spend a bit more time to see if one can easily push the hakyll yaml metadata through to pandoc, because I'd rather have references in `post.md` completely in yaml than having a separate bib file.

```haskell
addLinkCitations (Pandoc meta a) = Pandoc newMeta a
  where prevMap = unMeta meta
        newMap = M.insert "link-citations" (MetaBool True) prevMap
        newMeta = Meta newMap

myReadPandocBiblio :: ReaderOptions
                   -> Item CSL
                   -> Item Biblio
                   -> Item String
                   -> Compiler (Item Pandoc)
myReadPandocBiblio ropt csl biblio item = do
    -- Parse CSL file, if given
    style <- unsafeCompiler $ CSL.readCSLFile Nothing . toFilePath . itemIdentifier $ csl

    -- We need to know the citation keys, add then *before* actually parsing the
    -- actual page. If we don't do this, pandoc won't even consider them
    -- citations!
    let Biblio refs = itemBody biblio
    pandoc <- itemBody <$> readPandocWith ropt item
    let pandoc' = processCites style refs (addLinkCitations pandoc)

    return $ fmap (const pandoc') item

bibtexCompilerWith readerOpts writerOpts bibPath cslPath = do
  csl <- load cslPath
  bib <- load (fromFilePath bibPath)

  getResourceBody
    >>= myReadPandocBiblio readerOpts csl bib
    >>= return . writePandocWith writerOpts
```

**Comment by fiedlr** [comment]
@vjeranc Thank you so much for the fix, it works for me too! I do not really understand why it cannot be set by default. I think it's a must have for anyone trying to publish TeX articles with citations on their website in html, it would be really annoying to look them up by hand.

@jaspervdj Is there really no easy way how to incorporate this into the writerOpts?

**Comment by jaspervdj** [comment]
Hmm, I think the proposed change looks like a good default, do you want to send a PR to add this to `Hakyll.Web.Pandoc.Biblio`?
